domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -100,28 +114,38 @@ func (pm *pruningManager) UpdatePruningPointByVirtual(stagingArea *model.Staging
 	}
 
 	if !hasPruningPoint {
-		err = pm.savePruningPoint(stagingArea, pm.genesisHash)
+		hasGenesis, err := pm.blocksStore.HasBlock(pm.databaseContext, stagingArea, pm.genesisHash)
 		if err != nil {
 			return err
 		}
+
+		if hasGenesis {
+			err = pm.savePruningPoint(stagingArea, pm.genesisHash)
```

Comment:
Shouldn't this be done during initialization?

Reply:
This is done during init when you add genesis